logical/src/main/java/org/apache/drill/common/expression/fn/JodaDateValidator.java

```diff
@@ -0,0 +1,213 @@
+  public static final String JODA_FULL_ERA_NAME = "G";
+  public static final String JODA_NAME_OF_DAY = "E";
+  public static final String JODA_TIME_ZONE_NAME = "TZ";
+  public static final String JODA_HOUR_12_NAME = "h";
+  public static final String JODA_HOUR_12_OTHER_NAME = "h";
+  public static final String JODA_HOUR_24_NAME = "H";
+  public static final String JODA_MINUTE_OF_HOUR_NAME = "m";
+  public static final String JODA_SECOND_OF_MINUTE_NAME = "s";
+  public static final String JODA_MILLISECOND_OF_MINUTE_NAME = "S";
+  public static final String JODA_WEEK_OF_YEAR = "w";
+  public static final String JODA_MONTH = "MM";
+  public static final String JODA_HALFDAY = "aa";
+
+  static {
+    ansiToJodaMap.put(ANSI_FULL_NAME_OF_DAY, JODA_FULL_NAME_OF_DAY);
+    ansiToJodaMap.put(ANSI_DAY_OF_YEAR, JODA_DAY_OF_YEAR);
+    ansiToJodaMap.put(ANSI_DAY_OF_MONTH, JODA_DAY_OF_MONTH);
+    ansiToJodaMap.put(ANSI_DAY_OF_WEEK, JODA_DAY_OF_WEEK);
+    ansiToJodaMap.put(ANSI_NAME_OF_MONTH, JODA_NAME_OF_MONTH);
+    ansiToJodaMap.put(ANSI_ABR_NAME_OF_MONTH, JODA_ABR_NAME_OF_MONTH);
+    ansiToJodaMap.put(ANSI_FULL_ERA_NAME, JODA_FULL_ERA_NAME);
+    ansiToJodaMap.put(ANSI_NAME_OF_DAY, JODA_NAME_OF_DAY);
+    ansiToJodaMap.put(ANSI_TIME_ZONE_NAME, JODA_TIME_ZONE_NAME);
+    ansiToJodaMap.put(ANSI_HOUR_12_NAME, JODA_HOUR_12_NAME);
+    ansiToJodaMap.put(ANSI_HOUR_12_OTHER_NAME, JODA_HOUR_12_OTHER_NAME);
+    ansiToJodaMap.put(ANSI_HOUR_24_NAME, JODA_HOUR_24_NAME);
+    ansiToJodaMap.put(ANSI_MINUTE_OF_HOUR_NAME, JODA_MINUTE_OF_HOUR_NAME);
+    ansiToJodaMap.put(ANSI_SECOND_OF_MINUTE_NAME, JODA_SECOND_OF_MINUTE_NAME);
+    ansiToJodaMap.put(ANSI_MILLISECOND_OF_MINUTE_NAME, JODA_MILLISECOND_OF_MINUTE_NAME);
+    ansiToJodaMap.put(ANSI_WEEK_OF_YEAR, JODA_WEEK_OF_YEAR);
+    ansiToJodaMap.put(ANSI_MONTH, JODA_MONTH);
+    ansiToJodaMap.put(ANSI_HALFDAY_AM, JODA_HALFDAY);
+    ansiToJodaMap.put(ANSI_HALFDAY_PM, JODA_HALFDAY);
+  }
+
+  static {
+    ansiValuesForDeleting.add(SUFFIX_SP);
+    ansiValuesForDeleting.add(PREFIX_FM);
+    ansiValuesForDeleting.add(PREFIX_FX);
+    ansiValuesForDeleting.add(PREFIX_TM);
+  }
+
+  /**
+   * Replaces all ansi patterns from {@param pattern},
+   * available in ansiToJodaMap keys to jodaTime equivalents.
+   *
+   * @param pattern date pattern in ansi format
+   * @return date pattern with replaced patterns in joda format
+   */
+  public static String toJodaFormat(String pattern) {
+    String preparedString = deleteFromAnsi(pattern);
+    StringBuilder builder = new StringBuilder(preparedString);
+
+    int start = 0;    //every time search of ansi token in pattern will start from this index.
+    int minPos;       //min position of the longest ansi token
+    do {
+      //finds first value with max length
+      minPos = builder.length();
+      String firstMatch = null;
+      for (String ansiPattern : ansiToJodaMap.keySet()) {     //keys sorted in length decreasing
```

Comment:
Please add a comment explaining why the keys need to be in length decreasing maybe with an example. e.g. month and mon?